Add deploy option and some other changes.

TV4Fun opened the pull request:
This adds an option to upload the generated egg to a PyPI repository
via the deploy goal. Also some rewrites to save on code reuse and adding
support for a few different distribution types. Also fixed the install command.

Commit:
Add deploy option and some other changes.

This adds an option to upload the generated egg to a PyPI repository
via the deploy goal. Also some rewrites to save on code reuse and adding
support for a few different distribution types. Also fixed the install command.

Commit:
Attempt to log errors as errors

UltimateDogg commented on src/main/java/com/github/mojos/distribute/AbstractSetupCommandMojo.java:
```diff
@@ -0,0 +1,98 @@
+    @Override
+    public void execute() throws MojoExecutionException {
+        final File buildDirectory = Paths.get(project.getBuild().getDirectory(), "maven-python").toFile();
+        final String setupOutputCanonicalPath = project.getProperties().getProperty("python.distribute.plugin.setup.path");
```
can you default this to the old path (i think it was src/main/python/setup.py) for compatibility?

TV4Fun replied:
Done.

UltimateDogg commented on src/main/java/com/github/mojos/distribute/DeployMojo.java:
```diff
@@ -0,0 +1,57 @@
+package com.github.mojos.distribute;
```
can you add something to readme to describe this. Does it work with all distribution types, etc.

TV4Fun replied:
Done.

Commit:
Several improvements

Add support for bdist_dumb packaging, set default path for setup.py,
and add documentation of new features to readme file.

UltimateDogg commented:
thanks!
